exercises/bracket-push/src/example.c

```diff
@@ -0,0 +1,52 @@
+#include "bracket_push.h"
+#include <stdio.h>
+
+static bool is_matching(char a, char b)
+{
+   if (a == '{')
+      return b == '}';
+   if (a == '[')
+      return b == ']';
+   if (a == '(')
+      return b == ')';
+
+   return false;
+}
+
+static bool is_opening(const char c)
+{
+   return (c == '{') || (c == '[') || (c == '(');
+}
+
+static bool is_closing(const char c)
+{
+   return (c == '}') || (c == ']') || (c == ')');
+}
+
+/* https://stackoverflow.com/a/2718114/6049386 */
```

Comment:
Might be worth adding the date (year and month) here, and some reference to the MIT license SO use for contributions, which is compatible with the MIT license used here. SO can, and have changed licenses before; they used to use CC BY-SA, which (I'm not at all sure) I think is not directly compatible with MIT.

I'm sure this is probably overkill, but if we include attributions (we should, where needed) then we should do so precisely.

Reply:
`/* https://stackoverflow.com/a/2718114/6049386 , MIT,  Feb 2018*/`would probably be sufficient